blog/docs/articles/kafka-broker-with-isolated-data-plane.md

```diff
@@ -0,0 +1,176 @@
+
+```bash
+cat <<-EOF | kubectl apply -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kafka-broker-config
+  namespace: default
+data:
+  default.topic.partitions: "10"
+  default.topic.replication.factor: "1"
+  bootstrap.servers: "my-cluster-kafka-bootstrap.kafka:9092"
+EOF
+```
+
+We now create the broker, that uses the class `KafkaNamespaced` and also uses the configuration we just created above:
+
+```bash
+cat <<-EOF | kubectl apply -f -
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  annotations:
+    eventing.knative.dev/broker.class: KafkaNamespaced
+  name: broker-isolated-data-plane
+  namespace: default
+spec:
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: kafka-broker-config
+    namespace: default
+EOF
+```
+
+When we check the user namespace where the `Broker` object is created in, we will see the data plane pods created:
+
+```bash
+kubectl get pods -n default
+
+NAME                                       READY   STATUS    RESTARTS   AGE
+kafka-broker-dispatcher-8497dd6fb6-h8kdg   1/1     Running   0          15s
+kafka-broker-receiver-84ff47fcd9-cv8j8     1/1     Running   0          15s
+```
+
+Similar to the data plane deployments, we will also see different services used for the `Broker` objects. Thus, the addresses of the brokers will be using a different service than the `kafka-broker-ingress` service in `knative-eventing` namespace.
+
+```bash
+kubectl get brokers.eventing.knative.dev broker-isolated-data-plane -n default
+
+NAME                         URL                                                                                        AGE   READY   REASON
+broker-isolated-data-plane   http://kafka-broker-ingress.default.svc.cluster.local/default/broker-isolated-data-plane   37s   True
+```
+
+In isolated data plane mode, a new data plane is created per namespace and not per `Broker` object. If we create another `Broker` object in the same namespace, it will use the same data plane:
+
+```bash
+cat <<-EOF | kubectl apply -f -
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  annotations:
+    eventing.knative.dev/broker.class: KafkaNamespaced
+  name: other-broker-isolated-data-plane
+  namespace: default
+spec:
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: kafka-broker-config
+    namespace: default
+EOF
+```
+
+No new data plane pods are created:
+
+```bash
+kubectl get pods -n default
+
+NAME                                       READY   STATUS    RESTARTS   AGE
+kafka-broker-dispatcher-8497dd6fb6-h8kdg   1/1     Running   0          72s
+kafka-broker-receiver-84ff47fcd9-cv8j8     1/1     Running   0          72s
+```
+
+Finally, when **all** of the `Broker` objects with class `KafkaNamespaced` is deleted in the namespace, the data plane will be removed:
+
+```bash
+kubectl delete brokers.eventing.knative.dev -n default --all
```

Comment:
why `-n` and `-A` (or `--all`)?

Reply:
I wanted to show that when you delete all brokers in a namespace, data plane in that namespace will be deleted.

`k delete -A` means delete from all namespaces. 
`k delete -n default --all` means delete all brokers in `default` ns.